### PR TITLE
Use case and sample scripts

### PR DESCRIPTION
**Pull request by mkwoods927**
This PR contains a use case and corresponding sample script for posting Civis job status alerts to Slack. It also contains a sample script for querying a MYSQL database and saving the results in Google Sheets. The use case for this script is coming soon!

### CODE REVIEW

**Comment by shaunagm** [comment]
Left a bunch of small comments/suggestions on both the Civis->Slack use case and the doc. Let me know if you need clarification or if you think something's unnecessary. Overall I think this looks fantastic and I can't wait to share it with the community. 

I'll come back to review [mysql_to_googlesheets.py](https://github.com/move-coop/parsons/pull/630/files#diff-58615fd163b73099dfcadf62522012ccae01a268080721f082db20b29465b24b) later.

**Comment by mkwoods927** [comment]
Thanks @shaunagm ! I made the edits to add the use case to the side bar and added a link to the sample script in the use case. I'm not seeing other comments in there though, where were those made?

File: docs/use_cases/civis_job_status_slack_alert.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+===============
+Civis Job Status Slack Alert
+===============
+
+The Movement Cooperative uses Civis Platform to run the data syncs its members rely on.
+Civis users are able to monitor the status of these pipelines within the Civis interface, but not all stakeholders are Civis users and if a problem occurs folks have to navigate to another system, such as Slack or TMC's support ticket system, to flag the issue, collaborate on troubleshooting, and get updates on the resolution.
+To make this process more efficient, promote transparency, and facilitate collaboration, the TMC Engineering team wrote a script using Parsons to post Civis job and workflow statuses to an internal Slack channel.
```
[the remark this answers is not in the document]
Comment (mkwoods927) [reply docs/use_cases/civis_job_status_slack_alert.rst]: Thanks Shauna! I've added a bit more detail and mentioned that the original request came from Indivisible (with Jason's permission). Honestly I think Indivisible is the only member making use of this so far, but maybe when we have more folks we can put that number in!

File: useful_resources/sample_code/civis_job_status_slack_alert.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        return ':runner:'
+    else:
+        return ':shrug:'
+
+# Returns a Parsons table with workflow and job data from the specified Civis project.     
+def get_workflows_and_jobs(project_id):
+    project = client.projects.get(project_id)
+    
+    # Get workflow and the job data from the project
+    workflows = [dict(x) for x in project['workflows']]
```
[the remark this answers is not in the document]
Comment (mkwoods927) [reply useful_resources/sample_code/civis_job_status_slack_alert.py:49]: That change makes sense to me! I'll go ahead and make it!

File: useful_resources/sample_code/civis_job_status_slack_alert.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    
+    t = get_workflows_and_jobs(CIVIS_PROJECT).sort(columns=['state', 'name'])
+    
+    logger.info(f'Found {t.num_rows} jobs and workflows in {project_name} project.')
+    
+    # This is a list of strings we will build with each job's status
+    output_lines = []
+    
+    for run in t:
+        if run['state'] == 'succeeded' and run['object_type'] == 'workflow':
```
[the remark this answers is not in the document]
Comment (mkwoods927) [reply useful_resources/sample_code/civis_job_status_slack_alert.py:107]: Honestly I don't remember. I can adjust it!

File: useful_resources/sample_code/mysql_to_googlesheets.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+# TITLE is the name of the Google Sheets workbook the script will create.
+TITLE = 'sheet_title_here'
+# TAB_LABEL is the name of the tab where your query results will appear in Google Sheets.
+TAB_LABEL = 'tab_label_here'
+# QUERY is the SQL query we will run against the MYSQL database.
+QUERY = '''-- Enter SQL here'''
+
+# Function to add data to spreadsheet tab.
+# There is a limit to the number of calls per minute so we try k number of times
+def tryoverwrite(table,k,sheet_id,tab_index):
```
[the remark this answers is not in the document]
Comment (mkwoods927) [reply useful_resources/sample_code/mysql_to_googlesheets.py]: Done!

File: docs/use_cases/civis_job_status_slack_alert.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkwoods927) [reply docs/use_cases/civis_job_status_slack_alert.rst]: Let me know what you think of this!

**Comment by mkwoods927** [comment]
@shaunagm I saw part of the Circle CI test failed and I'm working through those issues! One failure is because I'm importing os in both scripts, but it's not used in the version I have here because we instruct folks to set environment variables on their own. Would you prefer I remove that or keep it in an add a line showing how to set it, like SLACK_API_KEY = os.environ['API_KEY'] for example?

**Comment by shaunagm** [comment]
You can go ahead and remove the ``import os``. You can add this comment for people who might be confused about environmental variables:

`# more on environmental variables: https://move-coop.github.io/parsons/html/use_cases/contribute_use_cases.html#sensitive-information`

File: docs/use_cases/civis_job_status_slack_alert.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaunagm) [reply docs/use_cases/civis_job_status_slack_alert.rst]: It looks great! I think the Civis <> Slack use case is ready to go.

**Comment by shaunagm** [comment]
Sounds great, @mkwoods927! If you want to split these up into two PRs so you can get the first use case/sample script merged you're welcome to.

**Comment by mkwoods927** [comment]
Ok @shaunagm all the tests are passing now, I think I just need an approval from you! It looks like there are also merge conflicts happening, can you help me resolve those?

**Comment by shaunagm** [comment]
Wonderful! I can fix the merge conflicts when I merge the PR, so don't worry about them.

Should I go ahead and merge, or did you want to wait until you've got the use case for the `mysql_to_googlesheets.py` ready? Should be fine either way.

**Review by shaunagm — APPROVED** [review]

**Comment by shaunagm** [comment]
Merged. Thank you so much for this @mkwoods927! I'm going to announce it in the Slack just as soon as the docs themselves are live (since currently our docs only update when a new version is pushed to PyPI).

**Pull request merged**